src/iterator.hpp

```diff
@@ -104,10 +106,15 @@ class key_iterator : public iterator_base<key_iterator> {
 
     [[nodiscard]] object_view operator*()
     {
+        static thread_local detail::object key;
         if (current_.first.empty()) {
             return {};
         }
-        return ddwaf_object_stringl_nc(&current_key_, current_.first.data(), current_.first.size());
+        return (key = {.parameterName = nullptr,
+                    .parameterNameLength = 0,
+                    .stringValue = current_.first.data(),
+                    .nbEntries = current_.first.size(),
+                    .type = DDWAF_OBJ_STRING});
```

Comment:
not pretty. I imagine this is going away with v2?

Reply:
To be fair it was ugly before as well but yeah this will be gone once the keys are separate objects.